pkg/group/group_integration_test.go

```diff
@@ -0,0 +1,196 @@
+	}
+
+	var groupName string
+	{
+		group, err := groupService.Create("test-group", "test-hostname.com", true)
+		require.NoError(t, err)
+		groupName = group.Name
+	}
+
+	t.Run("CreateGroup", func(t *testing.T) {
+		t.Parallel()
+
+		t.Run("Deployable", func(t *testing.T) {
+			requestBody := strings.NewReader(`{
+				"name": "deployable-test-group",
+				"hostname": "deployable-test-hostname.com",
+				"deployable": true
+			}`)
+
+			var group model.Group
+			client.PostJSON(t, "/groups", requestBody, &group)
+			require.Equal(t, "deployable-test-group", group.Name)
+			require.Equal(t, "deployable-test-hostname.com", group.Hostname)
+			require.Equal(t, true, group.Deployable)
+		})
+
+		t.Run("NonDeployable", func(t *testing.T) {
+			requestBody := strings.NewReader(`{
+				"name": "non-deployable-test-group",
+				"hostname": "non-deployable-test-hostname.com",
+				"deployable": false
+			}`)
+
+			var group model.Group
+			client.PostJSON(t, "/groups", requestBody, &group)
+			require.Equal(t, "non-deployable-test-group", group.Name)
+			require.Equal(t, "non-deployable-test-hostname.com", group.Hostname)
+			require.Equal(t, false, group.Deployable)
+		})
+	})
+
+	t.Run("AddUserToGroup", func(t *testing.T) {
+		t.Parallel()
+
+		t.Run("AddUserToGroup", func(t *testing.T) {
+			path := fmt.Sprintf("/groups/%s/users/%s", groupName, userId)
+
+			client.Do(t, http.MethodPost, path, nil, http.StatusCreated)
+		})
+
+		t.Run("AddUserToGroupNonExistingGroup", func(t *testing.T) {
+			path := fmt.Sprintf("/groups/%s/users/%s", "non-existing-group", userId)
+
+			response := client.Do(t, http.MethodPost, path, nil, http.StatusNotFound)
+
+			require.Equal(t, "group \"non-existing-group\" doesn't exist", string(response))
+		})
+
+		t.Run("AddNonExistingUserToGroup", func(t *testing.T) {
+			nonExistingUserId := uint(123)
+			_, err := userService.FindById(nonExistingUserId)
+			require.Error(t, err, "user already exists")
+			path := fmt.Sprintf("/groups/%s/users/%s", groupName, fmt.Sprintf("%d", nonExistingUserId))
+
+			response := client.Do(t, http.MethodPost, path, nil, http.StatusNotFound)
+
+			require.Equal(t, "failed to find user with id 123", string(response))
+		})
+	})
+
+	t.Run("RemoveUserFromGroup", func(t *testing.T) {
+		t.Parallel()
+
+		t.Run("RemoveUserFromGroup", func(t *testing.T) {
+			err := groupService.AddUser(groupName, user.ID)
+			require.NoError(t, err)
+			path := fmt.Sprintf("/groups/%s/users/%s", groupName, userId)
+
+			client.Do(t, http.MethodDelete, path, nil, http.StatusNoContent)
+		})
+
+		t.Run("RemoveUserFromNonExistingGroup", func(t *testing.T) {
+			path := fmt.Sprintf("/groups/%s/users/%s", "non-existing-group", userId)
+
+			response := client.Do(t, http.MethodPost, path, nil, http.StatusNotFound)
+
+			require.Equal(t, "group \"non-existing-group\" doesn't exist", string(response))
+		})
+
+		t.Run("RemoveNonExistingUserFromGroup", func(t *testing.T) {
+			nonExistingUserId := uint(123)
```

Comment:
We discussed this a while back and you did try it so you are not against it completely 😋 I would try to inline such vars as much as possible so the code becomes more succinct. I don't mind repetition in tests, but favor less clutter.

```go
response := client.Do(t, http.MethodDelete, "/groups/testgroup/99999", nil, http.StatusNotFound)

require.Equal(t, "failed to find user with id 99999", string(response))
```

As we assert on 404 and the `failed to find user` we are also certain that this is the cause. So there is no need to do the `userService.FindById` + `require.Error` check.